ephios/plugins/eventautoqualification/consequences.py

```diff
@@ -0,0 +1,85 @@
+import logging
+import operator
+from collections import Counter
+
+from django.db import transaction
+from django.db.models import F, Q
+from django.utils import timezone
+
+from ephios.core.consequences import QualificationConsequenceHandler
+from ephios.core.models import AbstractParticipation, LocalParticipation
+from ephios.plugins.eventautoqualification.models import EventAutoQualificationConfiguration
+
+logger = logging.getLogger(__name__)
+
+
+@transaction.atomic()
+def create_qualification_consequences(sender, **kwargs):
+    """
+    Creates
+    """
+    relevant_participations = (
+        AbstractParticipation.objects.filter(
+            state=AbstractParticipation.States.CONFIRMED,
+            shift__end_time__lt=timezone.now(),
+            shift__event__auto_qualification_config__isnull=False,
+        )
+        .filter(~Q(shift__event__auto_qualification_config__handled_for=F("id")))
+        .select_related("shift", "shift__event", "shift__event__auto_qualification_config")
+    )
```

Comment:
We figured it would be smart to offer a `participations_finished` signal from core.